### PR TITLE
Quizzes Schema & List UI

### PR DESCRIPTION
**Pull request by azooz2003-bit**
#127 

Notes:
- Exam-style quizzes aren't provided with the standard `getQuizzes` request, will have to investigate other requests that enable this.

Update: yup, there was another request

CleanShot 2024-12-15 at 19 06 19@2x

https://github.com/user-attachments/assets/96f27db7-e666-443a-a479-33a2b8c51449

### CODE REVIEW

File: CanvasPlusPlayground/Features/Quiz/CourseQuizzesView.swift
Comment (rahulon12) [inline CanvasPlusPlayground/Features/Quiz/CourseQuizzesView.swift:1]: Can you add screenshots?

File: CanvasPlusPlayground/Schema/Persistable/Quizzes/QuizQuestionType.swift [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+//
+//  Created by Abdulaziz Albahar on 12/14/24.
+//
+
+
+public enum QuizQuestionType: String, Codable, CaseIterable {
+    case calculatedQuestion = "calculated_question", essayQuestion = "essay_question", fileUploadQuestion = "file_upload_question", fillInMultipleBlanksQuestion = "fill_in_multiple_blanks_question",
+        matchingQuestion = "matching_question", multipleAnswersQuestion = "multiple_answers_question", multipleChoiceQuestion = "multiple_choice_question", multipleDropdownsQuestion = "multiple_dropdowns_question",
+        numericalQuestion = "numerical_question", shortAnswerQuestion = "short_answer_question", textOnlyQuestion = "text_only_question", trueFalseQuestion = "trueFalseQuestion"
+}
```
Comment (rahulon12) [inline CanvasPlusPlayground/Schema/Persistable/Quizzes/QuizQuestionType.swift:13]: Can you make this multiple lines?

File: CanvasPlusPlayground/Schema/Persistable/Quizzes/QuizType.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    var title: String {
+        switch self {
+        case .practiceQuiz: return "Practice Quizzes"
+        case .assignment: return "Assignment Quizzes"
+        case .gradedSurvey: return "Graded Surveys"
+        case .survey: return "Surveys"
+        case .quizzesNext: return "New Quizzes"
+        case .unknown: return "Unknown"
+        }
+    }
```
Comment (rahulon12) [inline CanvasPlusPlayground/Schema/Persistable/Quizzes/QuizType.swift:24]: nit: FYI, `switch` statements can evaluate to a value

```suggestion
    var title: String {
        return switch self {
        case .practiceQuiz: "Practice Quizzes"
        case .assignment: "Assignment Quizzes"
        case .gradedSurvey: "Graded Surveys"
        case .survey: "Surveys"
        case .quizzesNext: "New Quizzes"
        case .unknown: "Unknown"
        }
    }
```

Comment (rahulon12) [reply CanvasPlusPlayground/Schema/Persistable/Quizzes/QuizType.swift:24]: Actually don’t even need the `return` in this case

**Review by rahulon12 — APPROVED** [review]
Small nits; good to go once done

**Comment by rahulon12** [comment]
I see this includes changes from #103, so do we wanna close that?

**Comment by azooz2003-bit** [comment]
> I see this includes changes from #103, so do we wanna close that?

better as a separate PR tho, can u approve it?

**Pull request merged**